src/script/interpreter.cpp

```diff
@@ -1830,6 +1830,19 @@ uint256 ComputeTapleafHash(uint8_t leaf_version, Span<const unsigned char> scrip
     return (HashWriter{HASHER_TAPLEAF} << leaf_version << CompactSizeWriter(script.size()) << script).GetSHA256();
 }
 
+uint256 ComputeTapbranchHash(Span<const unsigned char> a, Span<const unsigned char> b)
+{
+    assert(a.size() == 32);
+    assert(b.size() == 32);
+    HashWriter ss_branch{HASHER_TAPBRANCH};
+    if (std::lexicographical_compare(a.begin(), a.end(), b.begin(), b.end())) {
+        ss_branch << a << b;
+    } else {
+        ss_branch << b << a;
+    }
```

Comment:
Somewhat related to this PR: changing the order of `a` and `b` isn't caught by unit tests, only functional tests. Should there be an additional unit test?

Reply:
Maybe you could sneak in this unit test in the PR?

<details>
<summary>Diff</summary>

```diff
diff --git a/src/test/script_tests.cpp b/src/test/script_tests.cpp
index 935194057..d04b01c1e 100644
--- a/src/test/script_tests.cpp
+++ b/src/test/script_tests.cpp
@@ -1820,4 +1820,12 @@ BOOST_AUTO_TEST_CASE(bip341_keypath_test_vectors)
 
 }
 
+BOOST_AUTO_TEST_CASE(compute_tapbranch)
+{
+    uint256 hash1 = uint256S("8ad69ec7cf41c2a4001fd1f738bf1e505ce2277acdcaa63fe4765192497f47a7");
+    uint256 hash2 = uint256S("f224a923cd0021ab202ab139cc56802ddb92dcfc172b9212261a539df79a112a");
+    uint256 result = uint256S("a64c5b7b943315f9b805d7a7296bedfcfd08919270a1f7a1466e98f8693d8cd9");
+    BOOST_CHECK_EQUAL(ComputeTapbranchHash(hash1, hash2), result);
+}
+
 BOOST_AUTO_TEST_SUITE_END()
```

</details>